src/service/common.ts

```diff
@@ -0,0 +1,142 @@
+import {CodeGeneratorResponse} from "google-protobuf/google/protobuf/compiler/plugin_pb";
+import {FileDescriptorProto, MethodDescriptorProto, ServiceDescriptorProto} from "google-protobuf/google/protobuf/descriptor_pb";
+import {ExportMap} from "../ExportMap";
+import {WellKnownTypesMap} from "../WellKnown";
+import {getFieldType, MESSAGE_TYPE} from "../ts/FieldTypes";
+import {filePathToPseudoNamespace, replaceProtoSuffix, getPathToRoot, normaliseFieldObjectName} from "../util";
+
+export function createFile(output: string, filename: string): CodeGeneratorResponse.File {
+  const file = new CodeGeneratorResponse.File();
+  file.setName(filename);
+  file.setContent(output);
+  return file;
+}
+
+type CallingTypes = {
+  requestType: string
+  responseType: string
+};
+
+function getCallingTypes(method: MethodDescriptorProto, exportMap: ExportMap): CallingTypes {
+  return {
+    requestType: getFieldType(MESSAGE_TYPE, method.getInputType().slice(1), "", exportMap),
+    responseType: getFieldType(MESSAGE_TYPE, method.getOutputType().slice(1), "", exportMap),
+  };
+}
+
+function isUsed(fileDescriptor: FileDescriptorProto, pseudoNamespace: string, exportMap: ExportMap) {
+  return fileDescriptor.getServiceList().some(service => {
+    return service.getMethodList().some(method => {
+      const callingTypes = getCallingTypes(method, exportMap);
+      const namespacePackage = pseudoNamespace + ".";
+      return (
+        callingTypes.requestType.indexOf(namespacePackage) === 0 ||
+        callingTypes.responseType.indexOf(namespacePackage) === 0
+      );
+    });
+  });
+}
+
+export type ImportDescriptor = {
+  readonly namespace: string
+  readonly path: string
+};
+
+export type RPCMethodDescriptor = {
+  readonly nameAsPascalCase: string,
+  readonly nameAsCamelCase: string,
+  readonly functionName: string,
+  readonly serviceName: string,
+  readonly requestStream: boolean
+  readonly responseStream: boolean
+  readonly requestType: string
+  readonly responseType: string
+};
+
+export class RPCDescriptor {
```

Comment:
No changes to the code in this file, just extracted what was common between grpcnode.ts and grpcweb.ts